src/processes/nix_list.rs

```diff
@@ -0,0 +1,163 @@
+use std::path::PathBuf;
+use sysinfo::{Process, ProcessRefreshKind, ProcessesToUpdate, System, UpdateKind};
+
+#[cfg(target_os = "macos")]
+use macos_visible_windows::macos_visible_windows;
+
+pub fn active_executables() -> Result<ProcessList> {
+    let mut executables: HashMap<PathBuf, ProcessInfo> = HashMap::new();
+    let visible = visible_windows()?;
+    let mut sys = System::new();
+    sys.refresh_processes_specifics(
+        ProcessesToUpdate::All,
+        true,
+        ProcessRefreshKind::nothing()
+            .with_exe(UpdateKind::OnlyIfNotSet)
+            .with_user(UpdateKind::OnlyIfNotSet),
+    );
+    for (pid, process) in sys.processes() {
+        // process.exe() will return empty path if there was an error while trying to read /proc/<pid>/exe.
+        if let Some(path) = process.exe() {
+            let pid = pid.as_u32();
+            let executable = path.to_path_buf();
+            match executables.entry(executable) {
+                Entry::Occupied(mut e) => {
+                    let process_info = e.get();
+                    if !process_info.is_visible && visible.contains(&pid) {
+                        e.get_mut().is_visible = true;
+                    }
+                }
+                Entry::Vacant(e) => {
+                    let executable = e.key().clone();
+                    // .file_name() returns `None` if the path terminates in `..`
+                    // We use the absolute path in such a case.
+                    let display_name = path
+                        .file_name()
+                        .unwrap_or(path.as_os_str())
+                        .to_string_lossy()
+                        .to_string();
+                    let is_system = is_system(process);
+                    let is_visible = visible.contains(&pid);
+                    e.insert(ProcessInfo {
+                        executable,
+                        display_name,
+                        is_visible,
+                        is_system,
+                    });
+                }
+            }
+        }
+    }
+    Ok(executables.into_values().collect())
+}
+
+pub fn visible_windows() -> Result<HashSet<PID>> {
+    #[cfg(target_os = "macos")]
+    return macos_visible_windows();
+
+    #[cfg(target_os = "linux")]
+    // Finding visible windows is less useful on Linux, where more applications tend to be CLI-based.
+    // So we skip all the X11/Wayland complexity.
+    return Ok(HashSet::new());
+}
+
+fn is_system(process: &Process) -> bool {
+    #[cfg(target_os = "macos")]
+    return process
+        .exe()
+        .map(|path| path.starts_with("/System/"))
+        .unwrap_or(false);
+
+    #[cfg(target_os = "linux")]
+    // process.user_id() returns 0 even if process is started using `sudo`
+    return process
+        .user_id()
+        .and_then(|uid| {
+            sysinfo::Uid::try_from(1000)
+                .ok()
+                .map(|uid_1000| uid < &uid_1000)
```

Comment:
This looks more complicated than it needs to be.

`sysinfo::Uid` implements `Deref<Target=u32>` so you can convert that into u32 and compare with 1000u32 directly, without needing to make a new `Uid` from 1000 to compare them as `Uid`s:

```
use std::ops::Deref;

fn main() {
    assert!(
        *sysinfo::System::new()
            .process(sysinfo::Pid::from_u32(1))
            .unwrap()
            .user_id()
            .unwrap()
            .deref() < 1000u32
    );
}
```

Reply:
thanks :)